test/e2e/validation_test.go

```diff
@@ -43,6 +45,35 @@ func testWorkerLabel(t *testing.T) {
 	}
 }
 
+// getWMCOVersion returns the version of the operator. This is sourced from the WMCO binary used to create the operator image.
+// This function will return an error if the binary is missing.
+func getWMCOVersion() (string, error) {
+	cmd := exec.Command(wmcoPath, "version")
+	out, err := cmd.Output()
+	if err != nil {
+		return "", errors.Wrapf(err, "error running %s", cmd.String())
+	}
+	// out is formatted like:
+	// ./build/_output/bin/windows-machine-config-operator version: "0.0.1+4165dda-dirty", go version: "go1.13.7 linux/amd64"
+	versionSplit := strings.Split(string(out), "\"")
+	if len(versionSplit) < 3 {
+		return "", fmt.Errorf("unexpected version output")
+	}
+	return versionSplit[1], nil
+}
+
+// testVersionAnnotation tests all nodes are annotated with the version of the currently deployed WMCO
+func testVersionAnnotation(t *testing.T) {
+	operatorVersion, err := getWMCOVersion()
+	require.NoError(t, err, "could not get WMCO version")
+	for _, node := range gc.nodes {
+		t.Run(node.GetName(), func(t *testing.T) {
+			require.Containsf(t, node.Annotations, nc.VersionAnnotation, "node %s missing version annotation", node.GetName())
```

Comment:
I thought I mentioned this before. We should not stop the test on the first failure. It will be a good data point for us if there is a scenario where one node has the correct version annotation while the other does not.

Reply:
We are not stopping at the first failure here, each node is checked within a separate subtest https://github.com/openshift/windows-machine-config-operator/pull/88/files#diff-1a876af15bc4507294c2bc7e5f81dc37R70 

When one node fails, its subtest fails and the rest of the nodes are tested with their subtests.

Reply:
You are right. I missed this too.